lib/mailparser.js

```diff
@@ -331,7 +331,7 @@ MailParser.prototype._processStateHeader = function(line) {
                 }
                 attachment.stream = this._currentNode.stream;
 
-                this.emit("attachment", attachment);
+                this.emit("attachment", attachment, this._currentNode.parentNode);
```

Comment:
@andris9 I've run into a problem with this, `this._currentNode.parentNode` is not necessary the mail object, it can be another part of a multipart email. We need to pass the last `.parentNode` leaf here, I'll send another pull request. Sorry for missing that